### PR TITLE
fix(v2): improve BlogPostFrontMatter schema validation

### PR DESCRIPTION
**Pull request by nam-hle**
## Motivation

From #4591.

## What the PR does

* Complete the BlogPostFrontMatter typing and the corresponding Joi schema validator in _packages/docusaurus-plugin-content-blog/src/blogFrontMatter.ts_
* Add unit tests
* Display friendly messages for wrong schemas:
image

* @slorber: Can you explain the sentence: _"try to use the frontmatter TS types in the theme?"_ ?

I found the after parsing the frontmatter use `gray-matter` package, the `date` field will be converted to a `Date object`, not a raw string. But we are treating it as a string. Currently there is no problem about it, but maybe furture.
 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Unit test.

### CODE REVIEW

**Review by nam-hle — COMMENTED** [review]
Thanks for reviewing. I add an annotation for deprecated fields and provide warning messages for them.

File: packages/docusaurus-utils-validation/src/validationUtils.ts [hunk 1]
```diff
@@ -110,15 +110,23 @@ export function validateFrontMatter<T>(
...
     });
   } catch (e) {
+    const frontMatterString = JSON.stringify(frontMatter, null, 2);
+    const errorDetails = (e as Joi.ValidationError).details;
+    const invalidFields = errorDetails.map(({path}) => path).join(', ');
+    const errorMessages = errorDetails
+      .map(({message}) => ` - ${message}`)
+      .join('\n');
+
+    logValidationBugReportHint();
```
[the remark this answers is not in the document]
Comment (nam-hle) [reply packages/docusaurus-utils-validation/src/validationUtils.ts:123]: Sorry but can you explain the using of env variable detailedly :D

**Comment by slorber** [comment]
Thanks, that looks great!

Disabling the warnings for now as our own blog posts are still using those frontmatters (content is shared between v1/v2 so we can't update the frontmatter)

**Comment by nam-hle** [comment]
Thanks. Now I can work on the docs frontmatter smoothly :D

**Pull request merged**